sklearn/cluster/_kmeans.py

```diff
@@ -617,8 +644,8 @@ class KMeans(TransformerMixin, ClusterMixin, BaseEstimator):
         Relative tolerance with regards to Frobenius norm of the difference
         in the cluster centers of two consecutive iterations to declare
         convergence.
-        It's not advised to set `tol=0` since convergence might never be
-        declared due to rounding errors. Use a very small number instead.
+        If set to 0, convergence will be declared when the labels are the
```

Comment:
If set to 0 or very close to zero, convergence will be declared ...

Reply:
I think this part is just not relevant anymore. When labels are the same, it implies that centers did not move so the norm of the diff is 0 <= tol. The label check is just a technicality for us to work around rounding errors.